### PR TITLE
Update and Reorg Enable Data Lineage for External Systems

### LINKED ISSUE #1466: Add Snowflake section to "Enable data lineage" doc (& other updates)
### Context

In discussing with @denimalpaca and @virajmparekh about an upcoming workshop with a customer about Open Lineage, Airflow/Astro, and Snowflake -- we recognized that:

- There is nothing about Snowflake and OL in our current docs
- There is no information about what lineage data we collect with the Snowflake Operator
- Our integration guides seem hidden in our overall doc, when it should be the most prominent

### Ask

- [ ] Review the introduction and overview to `Enable data lineage`
- [ ] Consider shrinking the data lineage architecture image. It takes up too much page real estate.
- [ ] Move `Retrieve your OpenLineage API Key` to an H2
- [ ] Consider moving the integration guides OUT of tabs and into H2s so that they're easier to see and link to
- [ ] Remove the `Astronomer` section/tab, as it's only there for older versions of Astro Runtime
- [ ] Add more links to OL docs as needed
- [ ] Create a new `Snowflake` tab or section based on @denimalpaca's doc (https://www.notion.so/astronomerio/Lineage-Collection-in-Snowflake-8ea974cbff1540f4b315367450f80b6b)
- [ ] Include a matrix of open lineage data we collect with Airflow/Astro (from Benji also)

@kwcanuck @jwitz I think we'll need my/other folks' guidance here so hold up before getting started on this issue

### PR DESCRIPTION
**Pull request by kwcanuck**
Resolves #1466 

@jwitz as discussed, I completed a preliminary reorg of the topic. Now to you for additional refinement. 

We seem to have a lot of duplication. [This](https://docs.astronomer.io/astro/data-lineage-concepts#docusaurus_skipToContent_fallback) is an unindexed conceptual topic. Also the [Integration section](https://deploy-preview-1473--gallant-neumann-56599d.netlify.app/learn/category/integrations) below **Learn** seems to contain similar information. Should we add a _Related documentation_ heading?

### CODE REVIEW

File: astro/set-up-data-lineage.md [hunk 1]
```diff
@@ -238,13 +176,10 @@ In your Spark application, set the following properties to configure your lineag
      .getOrCreate()                                             # Astronomer recommends using a meaningful namespace such as `spark-dev` or `spark-prod`.
    ```
 
-#### Verify
+### Verify
 
 To confirm that your setup is successful, run a Spark job after you save your configuration. After you run this model, open the **Lineage** tab in the Cloud UI and go to the **Runs** page. Your recent Spark job run appears in the table of most recent runs.
```
Comment (jwitz) [inline astro/set-up-data-lineage.md:182]:
```suggestion

## Integrate with Snowflake 

This guide outlines how to set up lineage collection for Snowflake.

## Setup

Like other [supported Airflow operators](data-lineage-support-and-compatibility.md#supported-airflow-operators), the [SnowflakeOperator](https://registry.astronomer.io/providers/snowflake/modules/snowflakeoperator) has full data lineage support and capabilities by default. As long as you connect to Snowflake through an Airflow connection, any task using the SnowflakeOperator emits lineage data about the Snowflake tables it queries.

### Verify 

After you run a DAG with the SnowflakeOperator, open the **Lineage** tab in the Cloud UI and go to the **Graph** page. The task using the SnowflakeOperator should appear as a run connected to a dataset. 

Click the dataset to view:

- The time that Snowflake was accessed.
- Tables that were accessed by upstream/ downstream tasks.
- The name and description for each column in the dataset.
- Data quality checks for the accessed tables. Data quality checks require integrating Great Expectations with OpenLineage and Airflow. See [Integrate with Great Expectations](#integrate-with-great-expectations) and [Orchestrate Great Expectations with Airflow](https://docs.astronomer.io/learn/airflow-great-expectations#docusaurus_skipToContent_fallback).

**Review by jwitz — COMMENTED** [review]
Thanks for updating the formatting on this @kwcanuck ! I added a suggestion with what I think we should include for the Snowflake integration. Feel free to update the suggestion however you want and I can take one final look before we merge

**Comment by kwcanuck** [comment]
@jwitz this PR is ready for final review.

**Comment by kwcanuck** [comment]
From @jwitz - "...1473 looks good to me, but I think @paolaperaza  really needs to review that because it has implications for the demo she mentioned was happening this week"

File: astro/set-up-data-lineage.md [hunk 2]
```diff
@@ -1,92 +1,121 @@
...
 
-To generate lineage graphs for your data pipelines, you first need to configure your data pipelines to emit lineage data. Because lineage data can be generated in all stages of your pipeline, you can configure pipeline components outside of Astro, such as dbt or Databricks, to emit lineage data whenever they're running a job. Coupled with lineage data emitted from your DAGs, Astro generates a lineage graph that can provide context to your data before, during, and after it reaches your Deployment.
+All Astro Deployments use the [OpenLineage Airflow library](https://openlineage.io/integration/apache-airflow/) (`openlineage-airflow`) to gather lineage data. The OpenLineage Airflow library is installed on [Astro Runtime](runtime-image-architecture.md) by default.
 
-## Lineage architecture
+There are two ways to emit lineage data:
```
Comment (jwitz) [inline astro/set-up-data-lineage.md:14]:
```suggestion
There are two ways to emit lineage data to Astro:
```

File: astro/set-up-data-lineage.md [hunk 3]
```diff
@@ -1,92 +1,121 @@
...
+- Run a task on Astro with a supported Airflow operator. These operators use extractors to automatically emit lineage data and don’t require additional configuration. See [Supported Airflow operators](data-lineage-support-and-compatibility.md#supported-airflow-operators).
+- Integrate OpenLineage with an external service, such as dbt or Apache Spark, to emit data lineage outside of an Airflow DAG or task. This might require additional configuration.
 
-![Diagram showing how lineage data flows to Astro](/img/docs/lineage-diagram.png)
+The data lineage graph in the Cloud UI shows lineage data that is emitted with both methods, including tasks or jobs that are not run on Astro. This graph can provide context to your data before, during, and after it reaches your Deployment. See [View data lineage](data-lineage.md).
```
Comment (jwitz) [inline astro/set-up-data-lineage.md:19]:
```suggestion
The data lineage graph in the Cloud UI shows lineage data that is emitted with both methods, including jobs that are not run on the Astro data plane. This graph can provide context to your data before, during, and after it reaches your Deployment. See [View data lineage](data-lineage.md).
```

File: astro/set-up-data-lineage.md [hunk 4]
```diff
@@ -1,92 +1,121 @@
...
 
-To generate lineage graphs for your data pipelines, you first need to configure your data pipelines to emit lineage data. Because lineage data can be generated in all stages of your pipeline, you can configure pipeline components outside of Astro, such as dbt or Databricks, to emit lineage data whenever they're running a job. Coupled with lineage data emitted from your DAGs, Astro generates a lineage graph that can provide context to your data before, during, and after it reaches your Deployment.
+All Astro Deployments use the [OpenLineage Airflow library](https://openlineage.io/integration/apache-airflow/) (`openlineage-airflow`) to gather lineage data. The OpenLineage Airflow library is installed on [Astro Runtime](runtime-image-architecture.md) by default.
```
Comment (jwitz) [inline astro/set-up-data-lineage.md:12]:
```suggestion
To store lineage data that is emitted from any system, a lineage backend is required. Astro Deployments use [OpenLineage](https://openlineage.io/integration/apache-airflow/) (`openlineage-airflow`) as a backend for gathering lineage data. The OpenLineage Airflow library is installed on [Astro Runtime](runtime-image-architecture.md) by default.
```

File: astro/set-up-data-lineage.md [hunk 5]
```diff
@@ -1,92 +1,121 @@
...
+- Run a task on Astro with a supported Airflow operator. These operators use extractors to automatically emit lineage data and don’t require additional configuration. See [Supported Airflow operators](data-lineage-support-and-compatibility.md#supported-airflow-operators).
+- Integrate OpenLineage with an external service, such as dbt or Apache Spark, to emit data lineage outside of an Airflow DAG or task. This might require additional configuration.
 
-![Diagram showing how lineage data flows to Astro](/img/docs/lineage-diagram.png)
+The data lineage graph in the Cloud UI shows lineage data that is emitted with both methods, including tasks or jobs that are not run on Astro. This graph can provide context to your data before, during, and after it reaches your Deployment. See [View data lineage](data-lineage.md).
 
-Configuring a system to send lineage data requires:
+## Integrate Astro and OpenLineage with external systems
```
Comment (jwitz) [inline astro/set-up-data-lineage.md:21]:
```suggestion
## Integrate Astro's lineage backend with external systems
```

File: astro/set-up-data-lineage.md [hunk 6]
```diff
@@ -1,92 +1,121 @@
...
 
-- Installing an OpenLineage backend to emit lineage data from the system.
-- Specifying your Organization's OpenLineage API endpoint to send lineage data to the Astro control plane.
+To store lineage data that is emitted from any system, a lineage backend is required. If you're an Astro customer, use Astro as a lineage backend instead of configuring a custom solution with a tool such as [Marquez](https://marquezproject.github.io/marquez/). To emit lineage data from Airflow tasks that run on Astro, Astro is automatically set as the lineage backend and no additional configuration is required.
 
-### Retrieve your OpenLineage API key
+To emit lineage data from an Airflow task that runs outside of Astro or from an external system that does not interact with Airflow:
```
Comment (jwitz) [inline astro/set-up-data-lineage.md:25]:
```suggestion
To emit lineage data from an Airflow task that runs outside of Astro or from an external system that does not interact with Airflow:
```

File: astro/set-up-data-lineage.md [hunk 7]
```diff
@@ -1,92 +1,121 @@
...
 
-To send lineage data from an external system to Astro, you must specify your Organization's OpenLineage API key in the external system's configuration.
+- Retrieve an OpenLineage API key from Astro. There is only one API key for each Organization.
+- Specify your OpenLineage API key in the external system.
```
Comment (jwitz) [inline astro/set-up-data-lineage.md:28]:
```suggestion
- Specify your OpenLineage API key in the external system. See the following integration guides for specific instructions.
```

File: astro/set-up-data-lineage.md [hunk 8]
```diff
@@ -1,92 +1,121 @@
...
 
-- Installing an OpenLineage backend to emit lineage data from the system.
-- Specifying your Organization's OpenLineage API endpoint to send lineage data to the Astro control plane.
+To store lineage data that is emitted from any system, a lineage backend is required. If you're an Astro customer, use Astro as a lineage backend instead of configuring a custom solution with a tool such as [Marquez](https://marquezproject.github.io/marquez/). To emit lineage data from Airflow tasks that run on Astro, Astro is automatically set as the lineage backend and no additional configuration is required.
 
-### Retrieve your OpenLineage API key
+To emit lineage data from an Airflow task that runs outside of Astro or from an external system that does not interact with Airflow:
 
-To send lineage data from an external system to Astro, you must specify your Organization's OpenLineage API key in the external system's configuration.
+- Retrieve an OpenLineage API key from Astro. There is only one API key for each Organization.
```
Comment (jwitz) [inline astro/set-up-data-lineage.md:27]:
```suggestion
- Retrieve your Organization OpenLineage API key from the Cloud UI. See [Retrieve an OpenLineage API key](#retrieve-an-openlineage-api-key).
```

File: astro/set-up-data-lineage.md [hunk 9]
```diff
@@ -1,92 +1,121 @@
...
+- Specify your OpenLineage API key in the external system.
+- Set Astro as your OpenLineage backend.
```
Comment (jwitz) [inline astro/set-up-data-lineage.md:29]:
```suggestion
```
This is not a step that users have to take.

**Review by jwitz — APPROVED** [review]
Approved with changes!

**Pull request merged**